addons/point_of_sale/static/src/js/Screens/PaymentScreen/PaymentScreen.js

```diff
@@ -21,6 +21,7 @@ odoo.define('point_of_sale.PaymentScreen', function (require) {
             useListener('send-payment-cancel', this._sendPaymentCancel);
             useListener('send-payment-reverse', this._sendPaymentReverse);
             useListener('send-force-done', this._sendForceDone);
+            this.currentOrder.on('rewards-updated', () => {this.render();});
```

Comment:
Might not work if pos_coupon is not installed.

Reply:
In ProductScreen, we listen to client changes and rerender. Perhaps you can use similar code for the PaymentScreen. Make sure the listener is removed when the component is unmounted.

Reply:
Yes but if we only render after the client changes the reward is not updated. We must wait for the 'rewards-updated' event to ensure that rewards are updated.

Reply:
I see. I guess we have to do it in two places then. Don't worry about calling render almost at the same time, they're scheduled by owl.

1. In point_of_sale PaymentScreen, we rerender on client change.
2. in pos_coupon, augment PaymentScreen and rerender on rewards-updated.
3. Make sure that you stop listening when unmounting.

What do you think?

Reply:
Yes that makes sense to me. 
Would this do the trick in the paymentScreen `this.env.pos.on('change:selectedClient', this.render, this);` to cover the client changes ? I'm not sure how this `change:selectedClient` works

Reply:
Yes. I think it will do the trick. `change:selectedClient` is a backbonejs-specific feature. Basically, the `pos` backbone object has a props called "selectedClient" such that when we call "pos.set('selectedClient')", the listeners react. In the case of ProductScreen, we rerender it when that props changes.